provider/acme/provider.go

```diff
@@ -130,7 +131,8 @@ func (p *Provider) Init(_ types.Constraints) error {
 	}
 
 	// Reset Account if caServer changed, thus registration URI can be updated
-	if p.account != nil && p.account.Registration != nil && !strings.HasPrefix(p.account.Registration.URI, p.CAServer) {
+	if p.account != nil && p.account.Registration != nil && isAccountMatchingCaServer(p.account.Registration.URI, p.CAServer) {
```

Comment:
It's should be `!isAccountMatchingCaServer(account.Registration.URI, a.CAServer)` instead of `isAccountMatchingCaServer(account.Registration.URI, a.CAServer)` no ?

Reply:
true, my mistake